src/TestUtils/src/Microsoft.Maui.IntegrationTests/Utilities/DotnetInternal.cs

```diff
@@ -34,6 +34,26 @@ public static bool Build(string projectFile, string config, string target = "",
 				binlogPath = Path.Combine(Path.GetDirectoryName(projectFile) ?? "", binlogName);
 			}
 
+			if (msbuildWarningsAsErrors)
```

Comment:
This section is what now causes the _template_ tests to treat MSBuild warnings as errors. There are two excluded warnings (see below, with justification). I originally had the _sample_ tests also do this, but our samples are not nearly as cleanly authored, and aren't as important, so the samples tests do NOT treat MSBuild warnings as errors.